pika/adapters/select_connection.py

```diff
@@ -27,20 +27,25 @@
 ERROR = 0x0008
 
 
-if pika.compat.PY2:
-    _SELECT_ERROR = select.error
-else:
-    # select.error was deprecated and replaced by OSError in python 3.3
-    _SELECT_ERROR = OSError
 
+_SELECT_ERRORS = (select.error, OSError, IOError)
 
-def _get_select_errno(error):
-    if pika.compat.PY2:
-        assert isinstance(error, select.error), repr(error)
-        return error.args[0]
-    else:
-        assert isinstance(error, OSError), repr(error)
-        return error.errno
+_SELECT_ERROR_CHECKERS = {
+    select.error : lambda e: e.args[0] == errno.EINTR,
+    OSError      : lambda e: e.errno == errno.EINTR,
+    IOError      : lambda e: e.errno == errno.EINTR,
+    }
```

Comment:
As for python 3.5 compatibility, the test (I wrote 'tests' before but there is only one test that exercise EINTR handling. my bad) should run except that EINTR exception does not get handled. cf. https://www.python.org/dev/peps/pep-0475/
I introduced pika.compat.EINTR_IS_EXPOSED flag and asserts on _is_resumable_mock.call_count accordingly.